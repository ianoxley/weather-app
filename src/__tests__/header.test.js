import React from 'react';
import { shallow } from 'enzyme';
import Header from '../header';

describe('Header', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Header
        location='Newcastle-upon-Tyne'
        currentTemperature='17'
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
