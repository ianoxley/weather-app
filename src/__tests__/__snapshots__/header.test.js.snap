// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    currentTemperature="17"
    location="Newcastle-upon-Tyne"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="app-title"
>
          Weather for 
          Newcastle-upon-Tyne
</h1>,
        <p
          id="current-temp"
>
          17
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Weather for ",
            "Newcastle-upon-Tyne",
          ],
          "className": "app-title",
        },
        "ref": null,
        "rendered": Array [
          "Weather for ",
          "Newcastle-upon-Tyne",
        ],
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "17",
          "id": "current-temp",
        },
        "ref": null,
        "rendered": "17",
        "type": "p",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="app-title"
>
            Weather for 
            Newcastle-upon-Tyne
</h1>,
          <p
            id="current-temp"
>
            17
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Weather for ",
              "Newcastle-upon-Tyne",
            ],
            "className": "app-title",
          },
          "ref": null,
          "rendered": Array [
            "Weather for ",
            "Newcastle-upon-Tyne",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "17",
            "id": "current-temp",
          },
          "ref": null,
          "rendered": "17",
          "type": "p",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
